fix(docs-infra): a couple of minor fixes for angular.io

gkalpak opened the pull request:
☝️ (See individual commits for details.)

mary-poppins commented:
You can preview 690b344 at https://pr42584-690b344.ngbuilds.io/.

mary-poppins commented:
You can preview 566a40f at https://pr42584-566a40f.ngbuilds.io/.

jessicajaniuk reviewed (approved):
LGTM :cookie:

reviewed-for: size-tracking

atscott reviewed (approved):
reviewed-for: size-tracking

dario-piotrowicz commented on aio/src/app/app.component.html:
```diff
@@ -28,13 +28,13 @@
-      <a href="https://twitter.com/angular" title="Twitter" aria-label="Angular on twitter">
+      <a mat-icon-button href="https://twitter.com/angular" title="Twitter" aria-label="Angular on twitter">
         <mat-icon svgIcon="logos:twitter"></mat-icon>
       </a>
-      <a href="https://github.com/angular/angular" title="GitHub" aria-label="Angular on github">
+      <a mat-icon-button href="https://github.com/angular/angular" title="GitHub" aria-label="Angular on github">
         <mat-icon svgIcon="logos:github"></mat-icon>
       </a>
-      <a href="https://youtube.com/angular" title="YouTube" aria-label="Angular on YouTube">
+      <a mat-icon-button href="https://youtube.com/angular" title="YouTube" aria-label="Angular on YouTube">
```
the icons seem to have lost they hover effect (https://pr42584-566a40f.ngbuilds.io/.)

is this wanted?

dario-piotrowicz replied:
ah yeah I can see the hover effect removed from the `a`s below, sorry

petebacondarwin commented on goldens/size-tracking/aio-payloads.json:
```diff
@@ -3,7 +3,7 @@
       "uncompressed": {
         "runtime-es2017": 4619,
-        "main-es2017": 456795,
+        "main-es2017": 456863,
```
Interestingly, while this change incurs a cost in JS, there is actually a decrease in the CSS that is downloaded (and must be parsed and applied by the browser). So it is not that clear what the net result is here.

I wonder if there may be value in also tracking the size of the `styles.css` file?

gkalpak replied:
Yeah, I was thinking about that one too. I added a commit to start tracking the CSS size as well: https://github.com/angular/angular/pull/42584/commits/d05b95a17d18d3b3825d9ba672301ddc836e331e

petebacondarwin reviewed (approved):
LGTM! Thanks for keeping things tight!

There is a typo in the last commit:

> This commit fixes the issue ~~be~~ _by_ also setting

petebacondarwin reviewed (approved):
Reviewed-for: size-tracking

mary-poppins commented:
You can preview b066d9c at https://pr42584-b066d9c.ngbuilds.io/.

mary-poppins commented:
You can preview dcd42c9 at https://pr42584-dcd42c9.ngbuilds.io/.

gkalpak committed:
ci: start tracking CSS payload sizes for angular.io

Previously, we only tracked the sizes of the eagerly loaded JS bundles.
However, the CSS styles (which also have a non-negligible size) must
also be downloaded and parsed by the browser, thus affecting the initial
rendering time.

This commit starts tracking the CSS styles payload sizes.
(Originally discussed in
https://github.com/angular/angular/pull/42584#discussion_r653787961)

gkalpak committed:
fix(docs-infra): remove redundant `MatIcon` element

Previously, we included a dummy `MatIcon` element in `AppComponent` in
order to ensure that the `Material Icons` font would be requested (and
thus cached by the ServiceWorker) on every navigation. However, #41129
introduced the `ThemeToggleComponent`, which will be present on all
pages (since it is located in the top-menu) and relies on the
`Material Icons` font. Therefore, the work-around for loading the
`Material Icons` font is no longer necessary.

This commit removes the now redundant `MatIcon` from `AppComponent`.

gkalpak committed:
fix(docs-infra): make `NotificationComponent` elements non-focusable when hidden

Previously, the `NotificationComponent` would be hidden by reducing its
height to `0`. This allowed for a smoother hide animation when closing
the notification, but left the component's interactive elements
focusable via keyboard navigation. This was confusing for users, because
the focused elements would not be visible on the page.

This commit fixes the issue be also setting the `display` CSS property
to `none` when the `NotificationComponent` is hidden, thus ensuring that
its contents are not focusable via keyboard navigation.
(This does not affect the hide animation, since the `display` style is
not animatable and is only applied at the end of the animation.)

gkalpak committed:
refactor(docs-infra): remove redundant CSS rule

Since we now use SVG for external link icons, there are no `<img>`
elements inside the anchor elements. So, the CSS rule does not match any
element and can be removed.

gkalpak committed:
fix(docs-infra): convert external links to `MatIconButton`s

This commit changes the anchor elements used for external links to
`MatIconButton`s. While the appearance remains the same (with the
exception of hover/focus styles), this better aligns the styling of
external link icons with other nearby icon buttons (i.e. the theme
toggle) and alows as to simplify the CSS for external links (since much
of their styling is handled by Angular Material).

mary-poppins commented:
You can preview 3fb0ee8 at https://pr42584-3fb0ee8.ngbuilds.io/.
